lib-multisrc/mangahub/src/eu/kanade/tachiyomi/multisrc/mangahub/MangaHub.kt

```diff
@@ -104,9 +105,32 @@ abstract class MangaHub(
             baseUrl.toHttpUrl()
         }
 
-        // Clear key cookie
-        val cookie = Cookie.parse(url, "mhub_access=; Max-Age=0; Path=/")!!
-        client.cookieJar.saveFromResponse(url, listOf(cookie))
+        val regex = Regex("mhub_access=([^;]+)")
+        val oldKey = client.cookieJar
+            .loadForRequest(baseUrl.toHttpUrl())
+            .firstOrNull { it.name == "mhub_access" && it.value.isNotEmpty() }?.value
+
+        for (i in 1..2) {
+            // Clear key cookie
+            val cookie = Cookie.parse(url, "mhub_access=; Max-Age=0; Path=/")!!
+            client.cookieJar.saveFromResponse(url, listOf(cookie))
+
+            // We try requesting again with param if the first one fails
+            val query = if (i == 2) "?reloadKey=1" else ""
+
+            try {
+                val response = client.newCall(GET("$url$query", headers)).execute()
+                val returnedKey = response.headers["set-cookie"]?.let { regex.find(it)?.groupValues?.get(1) }
+
+                if (returnedKey == oldKey) {
+                    continue // Site gave an old key so we try again
+                }
+
+                break; // Break out of loop since we got an allegedly valid API key
```

Comment:
I think this can be simplified somewhat:
```suggestion
                // Break if we got an allegedly valid API key, otherwise try again
                if (returnedKey != oldKey) break
```